src/backend/distributed/commands/dependencies.c

```diff
@@ -173,13 +173,15 @@ GetDependencyCreateDDLCommands(const ObjectAddress *dependency)
 		{
 			const char *schemaDDLCommand = CreateSchemaDDLCommand(dependency->objectId);
 
+			List * grantDDLCommands = GrantOnSchemaDDLCommands(dependency->objectId);
+
 			if (schemaDDLCommand == NULL)
 			{
 				/* no schema to create */
 				return NIL;
 			}
 
-			return list_make1((void *) schemaDDLCommand);
+			return lcons((void *) schemaDDLCommand, grantDDLCommands);
```

Comment:
it is a bit unusual to see this. It does work, but is more readable if we would make a list earlier in the block and `lconcat` the grant statements to it.